types/fs-base-ui/html_ui/JS/common.d.ts

```diff
@@ -0,0 +1,1102 @@
+        unFocusByKeys(): void;
+    }
+
+    let g_checkComponentsTimeout: number;
+
+    class TemplateElement extends UIElement {
+        static call(obj: Element, fn: Function, ...args: any[]): void;
+        static callNoBinding(obj: Element, fn: () => void): void;
+
+        created: boolean;
+        instantciatePopupToolbar: () => void;
+        callbackCreated: () => void;
+        get templateID(): string;
+        Instanciate(): null | void;
+        querySelectorH(selector: string): Element | null;
+        appendContent(element: Node): void;
+        onResourceLoaded(e: CustomEvent<any>): void;
+        convertPath(path: string): string;
+    }
+
+    class UIMarquee extends UIElement {
+        constructor(...args: any[]);
+        updateScrollBehaviour: () => void;
+        updateScrollAnimation: () => void;
+        startScrollAnimation: () => void;
+        stopScrollAnimation: () => void;
+        needsTooltip(): boolean;
+        needsEllipsis(): boolean;
+        noSizeCheck(): boolean;
+        set manual(value: boolean);
+        get manual(): boolean;
+        scrollAnimation(): UITimeline;
+    }
+
+    class LabelizeElement extends UIElement {
+        constructor(...args: any[]);
+        updateText: () => void;
+        set key(value: string | null);
+        get key(): string | null;
+    }
+
+    class UIImageElement extends TemplateElement {
+        onImageLoaded: () => void;
+        onImageError: () => void;
+        updateBackground: () => void;
+        get transition(): boolean;
+        LoadContentURL(srcImage: string, urlContent: string, contentTag: string, contentData: any): void;
+        set src(value: string);
+        get src(): string;
+        clear(): void;
+        set transitionTime(value: number);
+        get transitionTime(): number;
+    }
+
+    namespace ViewListener {
+        class ViewListener {
+            constructor(name: string);
+            CheckCoherentEvent: (listenerName: string, ...args: any[]) => void;
+            unregister: () => void;
+            onEventToAllSubscribers: (eventName: string, ...args: any[]) => void;
+            onGlobalEvent(eventName: string, ...args: any[]): void;
+            off(name: string, callback: (...args: any[]) => void, context: any): void;
+            on(name: string, callback: (...args: any[]) => void, context: any): void;
+            trigger(name: string, ...args: any[]): void;
+            triggerToAllSubscribers(event: any, ...args: any[]): void;
+        }
+
+        class ViewListenerMgr {
+            OnListenerRegistered: (name: any) => void;
+            onRegister(name: string, viewListener: ViewListener): void;
+            getListenerByName(name: string): ViewListener;
+            onUnregister(name: string, viewListener: ViewListener, force?: boolean): void;
+        }
+
+        let g_ViewListenersMgr: ViewListenerMgr;
+    }
+
+    function RegisterViewListenerT<T>(name: string, callback: (() => void) | void, type: new() => T,
```

Comment:
`callback` is optional.

Reply:
I have declared it optional with the `void` type. An optional parameter before a non-optional parameter cannot be defined in TypeScript (and I doubt it's legal in JavaScript).